protos/sources/vega/api/v1/core.proto

```diff
@@ -343,3 +348,84 @@ message LastBlockHeightResponse {
   // the network chain id from which the block comes from
   string chain_id = 8;
 }
+
+// A request to retrieve the spam statistics of a party for the given epoch
+message GetSpamStatisticsRequest {
+  // Id for the party whose statistics are requested
+  string party_id = 1;
+}
+
+// The statistics for a given spam policy
+message SpamStatistic {
+  // The current transaction count received from the party during this epoch for this policy
+  uint64 count_for_epoch = 1;
+  // The maximum number of transactions allowed for this policy in an epoch
+  uint64 max_for_epoch = 2;
+  // If blocked the timestamp when the party will be unblocked as RFC3339Nano
+  optional string banned_until = 4;
+}
+
+// The voting statistics by proposal for a given party for the current epoch
+message VoteSpamStatistics {
+  // List of statistics for proposals voted on by the party
+  repeated VoteSpamStatistic statistics = 1;
+  // If blocked the timestamp when the party will be unblocked as RFC3339Nano
+  optional string banned_until = 2;
+}
+
+// The vote statistics for the voting spam policies
+// which are calculated as a ratio of the total votes
+// that have been rejected.
+message VoteSpamStatistic {
+  string proposal = 1;
+  // The current vote count received from the party for the given proposal during this epoch
+  uint64 count_for_epoch = 2;
+  // The maximum number of votes per proposal allowed in an epoch
+  uint64 max_for_epoch = 3;
+  // The effective minimum number of tokens required to vote on the proposal
+  string min_tokens_required = 4;
+}
+
+// The PoW state for a given block
+message PoWBlockState {
+  // Block height for the current Proof of Work state statistics
+  uint64 block_height = 1;
+  // Hash of the current block
+  string block_hash = 2;
+  // Total number of transactions seen in the block
+  uint64 transactions_seen = 3;
+  // The minimum expected difficulty for the next transaction in the block
+  // This is only provided if increasing difficulty is required
+  optional uint64 expected_difficulty = 4;
```

Comment:
Why not return always return it so we don't have to compute it on the wallet side ? That would save us from implementing the algorithm inside the wallet. And consolidating the state would be much simpler.

Reply:
I'm not sure what you're saying. if it's nil it means you can't send any more transaction on this block. If it is not nil, then this is the exact (minimum) difficulty you need to use in the wallet.

Reply:
Right my question was confusing, but your response is what was I was looking for.

It makes sense, and fits what I would like.

However, the comment needs to be rework, bale cause it basically says the value is present only when the difficulty is increased which is not what you explained, right ?

Reply:
I would suggest rephrasing the comment as:
This is the minimum required difficulty for the next transaction submitted on this block if it is possible to submit more transactions on this block, otherwise nil.

Reply:
As it was explained to me, when the increaseDifficulty param is false, then there is no difficulty calculation, so there's no value for the expected difficulty.